TwitchClient/GamesListViewController.swift

```diff
@@ -1,108 +0,0 @@
-//
-//  GamesListViewController.swift
-//  TwitchClient
-//
-//  Created by Ahmed kotb on 4/7/17.
-//  Copyright © 2017 Ahmed Kotb. All rights reserved.
-//
-
-import UIKit
-import SVProgressHUD
-
-class GamesListViewController: UIViewController, WebserviceManagerDelegate, UITableViewDelegate, UITableViewDataSource {
-
-    
-    @IBOutlet weak var gamesTableView: UITableView!
-    
-    
-    var games = [Game]()
-   
-    var selectedGame = Game()
-    
-    override func viewDidLoad() {
-        super.viewDidLoad()
-
-        // Do any additional setup after loading the view.
-        
-        self.gamesTableView.isHidden = true
-        SVProgressHUD.show()
-        
-        self.getTopGames()
-        
-    }
-   
-    func getTopGames()
```

Comment:
A VC should only be concerned with displaying data and handling event. Talking to the network manager shouldn't be its job. It seems to me you only ever worked with MVC. I would suggest you first try to migrate this to MVVM, while not a perfect architecture, it's much better than MVC.

Topics: Massive View Controller, MVVM in Swift, single responsibility principle in SOLID, SOLID in general.